src/sql/workbench/browser/enablePreviewFeatures.ts

```diff
@@ -4,8 +4,24 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { AbstractEnablePreviewFeatures } from 'sql/workbench/common/enablePreviewFeatures';
+import { IStorageService } from 'vs/platform/storage/common/storage';
+import { INotificationService } from 'vs/platform/notification/common/notification';
+import { IHostService } from 'vs/workbench/services/host/browser/host';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
 
 export class BrowserEnablePreviewFeatures extends AbstractEnablePreviewFeatures {
+
+	constructor(
+		@IStorageService storageService: IStorageService,
+		@INotificationService notificationService: INotificationService,
+		@IHostService hostService: IHostService,
+		@IConfigurationService configurationService: IConfigurationService
+	) {
+		super(storageService, notificationService, hostService, configurationService);
+
+		this.handleTelemetryOptOut();
```

Comment:
Seems like an odd choice for a name given that it appears to have nothing to do with telemetry? Any idea what's going on with that?

Reply:
Ops, this entire feature is modeled off of vscode' telemetry output contribution, so this refactor was also modeled after that. I'll rename.